botocore/signers.py

```diff
@@ -130,3 +132,23 @@ def get_auth(self, signing_name, region_name, signature_version=None):
             auth = cls(**kwargs)
             self._cache[key] = auth
             return auth
+
+    def generate_url(self, request_dict, expires_in=None, region_name=None):
```

Comment:
Note it may be a little weird that we use request dict instead of a request object, but once we refactor the auth class to accept request_dict, we can change the ``sign()`` method to the style of the ``generate_url()`` method.